modules/justIdSystem.js

```diff
@@ -0,0 +1,187 @@
+   * @returns {{justId:string}}
+   */
+  decode(value) {
+    utils.logInfo(LOG_PREFIX, 'decode', value);
+    const justId = value && value.uid;
+    return justId && {justId: justId};
+  },
+
+  /**
+   * performs action to obtain id and return a value in the callback's response argument
+   * @function
+   * @param {SubmoduleConfig} config
+   * @param {ConsentData} consentData
+   * @param {(Object|undefined)} cacheIdObj
+   * @returns {IdResponse|undefined}
+   */
+  getId(config, consentData, cacheIdObj) {
+    utils.logInfo(LOG_PREFIX, 'getId', config, consentData, cacheIdObj);
+
+    return {
+      callback: function(cbFun) {
+        try {
+          utils.logInfo(LOG_PREFIX, 'fetching uid...');
+
+          var configWrapper = new ConfigWrapper(config);
+
+          var uidProvider = configWrapper.isAdvancedMode()
+            ? new AdvancedUidProvider(configWrapper, consentData, cacheIdObj)
+            : new BasicUidProvider(configWrapper);
+
+          uidProvider.getUid(justId => {
+            if (utils.isEmptyStr(justId)) {
+              utils.logError(LOG_PREFIX, 'empty uid!');
+              cbFun();
+              return;
+            }
+            cbFun({uid: justId});
+          }, err => {
+            utils.logError(LOG_PREFIX, 'error during fetching', err);
+            cbFun();
+          });
+        } catch (e) {
+          utils.logError(LOG_PREFIX, 'Error during fetching...', e);
+        }
+      }
+    };
+  }
+};
+
+export const ConfigWrapper = function(config) {
+  this.getConfig = function() {
+    return config;
+  }
+
+  this.getMode = function() {
+    return (params().mode || DEFAULT_MODE).toUpperCase();
+  }
+
+  this.getPartner = function() {
+    return params().partner || DEFAULT_PARTNER;
+  }
+
+  this.isAdvancedMode = function() {
+    return this.getMode() === MODE_ADVANCED;
+  }
+
+  this.getAtmVarName = function() {
+    return params().atmVarName || DEFAULT_ATM_VAR_NAME;
+  }
+
+  this.getUrl = function() {
+    const u = params().url || DEFAULT_URL;
+    const url = new URL(u);
+    url.searchParams.append('sourceId', this.getPartner());
+    return url.toString();
+  }
+
+  function params() {
+    return config.params || {};
+  }
+}
+
+const AdvancedUidProvider = function(configWrapper, consentData, cacheIdObj) {
+  const url = configWrapper.getUrl();
+
+  this.getUid = function(idCallback, errCallback) {
+    const scriptTag = jtUtils.createScriptTag(url);
+
+    scriptTag.addEventListener('justIdReady', event => {
+      utils.logInfo(LOG_PREFIX, 'received justId', event);
+      idCallback(event.detail && event.detail.justId);
+    });
+
+    scriptTag.onload = () => {
+      utils.logInfo(LOG_PREFIX, 'script loaded', url);
+      scriptTag.dispatchEvent(new CustomEvent('prebidGetId', { detail: { config: configWrapper.getConfig(), consentData: consentData, cacheIdObj: cacheIdObj } }));
+    };
+
+    scriptTag.onerror = errCallback;
+
+    document.head.appendChild(scriptTag);
```

Comment:
You must use https://github.com/prebid/Prebid.js/blob/804295aa2dae0484d67891b73fcfc401ef8244f1/src/adloader.js#L21

Please rename 'Advanced' mode to something that is more descriptive. Advanced mode implies it works better but the most advanced or full integration would be a pub loading your external javascript externally from this module and this module would just look for that library existing and grab from on page js or local storage, as many others do. 
Your "Advanced" mode is really just lazy mode, where publishers have one button full integration, but I understand that isnt a good name either.

Reply:
Perhaps mode "combined" with the definition being something like 'this mode combines our js library with prebid for ease of integration; prebid.org provides no warranty to its contents.'

Reply:
Finally your script does some odd things like pulling the interest cohort, so be aware that once you open it up, it still would be subject  to review. You might be better off removing the default location altogether. Contents at time of this comment: 

`try{
(function() {
var GET_ID;(()=>{"use strict";var t={d:(e,n)=>{for(var r in n)t.o(n,r)&&!t.o(e,r)&&Object.defineProperty(e,r,{enumerable:!0,get:n[r]})},o:(t,e)=>Object.prototype.hasOwnProperty.call(t,e),r:t=>{"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(t,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(t,"__esModule",{value:!0})}},e={};function n(t){var e=window.atob(t.replace(/_/g,"/").replace(/-/g,"+").substring(0,16));return 100*((63&e.charCodeAt(5))*Math.pow(2,30)+(e.charCodeAt(6)<<22|e.charCodeAt(7)<<14|e.charCodeAt(8)<<6|e.charCodeAt(9)>>>2))}function r(t,e){for(var n in e)e.hasOwnProperty(n)&&(t[n]=t[n]||e[n])}function o(t,e,n=""){for(var r in t=new URL(t),e)if(e.hasOwnProperty(r)){var i=e[r];p(i)?t=o(t,i,n+r+"."):(Array.isArray(i)?i:[i]).filter((t=>t)).forEach((e=>t.searchParams.append(n+r,e)))}return t}function i(t,e,n,r){if(t&&e&&n){var o=new Date;o.setTime(o.getTime()+1e3*n);var i=t+"="+e+";expires="+o.toUTCString()+";path=/";r&&(i=i+";domain="+r),document.cookie=i}}function u(t){const e=("; "+document.cookie).split("; "+t+"=");if(2===e.length)return e.pop().split(";").shift()}function c(t,e){return new URL(t).searchParams.get(e)}function s(){try{return window.top.location.href}catch(t){if(window.parent==window.top)return document.referrer}}function a(t,e){return toString.call(t)==="[object "+e+"]"}function f(t){return a(t,"Function")||a(t,"AsyncFunction")}function l(t){return a(t,"String")}function p(t){return a(t,"Object")}t.r(e),t.d(e,{init:()=>ut});const h={};var d=!1;function v(t,e){var n=y(t);if(n)return function(t,r,o,i){const u=Math.random()+"",c={};c[e]={command:t,parameter:i,version:r,callId:u},h[u]={callId:u,callback:o,removeAfterReturn:"addEventListener"!==t},n.postMessage(c,"*"),d||(window.addEventListener("message",w,!1),d=!0)}}function w(t){var e={};try{e=l(t.data)?JSON.parse(t.data):t.data}catch(t){}m(e.__tcfapiReturn),m(e.__cmpReturn)}function m(t){if(t&&t.callId){var e=h[t.callId];e&&f(e.callback)&&e.callback(t.returnValue,t.success),function(t){t.removeAfterReturn&&(delete h[t.callId],0==Object.keys(h).length&&(window.removeEventListener("message",w),d=!1))}(e)}}function y(t,e){var n;e||(e=window);try{n=e.frames[t]}catch(t){return window.top}return n?e:e!=window.top?y(t,e.parent):void 0}var b=function(){function t(t){void 0===t&&(t=5e3),this.timeout=t}return t.prototype.getConsents=function(){return t=this,e=void 0,i=function(){var t=this;return function(t,e){var n,r,o,i,u={label:0,sent:function(){if(1&o[0])throw o[1];return o[1]},trys:[],ops:[]};return i={next:c(0),throw:c(1),return:c(2)},"function"==typeof Symbol&&(i[Symbol.iterator]=function(){return this}),i;function c(i){return function(c){return function(i){if(n)throw new TypeError("Generator is already executing.");for(;u;)try{if(n=1,r&&(o=2&i[0]?r.return:i[0]?r.throw||((o=r.return)&&o.call(r),0):r.next)&&!(o=o.call(r,i[1])).done)return o;switch(r=0,o&&(i=[2&i[0],o.value]),i[0]){case 0:case 1:o=i;break;case 4:return u.label++,{value:i[1],done:!1};case 5:u.label++,r=i[1],i=[0];continue;case 7:i=u.ops.pop(),u.trys.pop();continue;default:if(!((o=(o=u.trys).length>0&&o[o.length-1])||6!==i[0]&&2!==i[0])){u=0;continue}if(3===i[0]&&(!o||i[1]>o[0]&&i[1]<o[3])){u.label=i[1];break}if(6===i[0]&&u.label<o[1]){u.label=o[1],o=i;break}if(o&&u.label<o[2]){u.label=o[2],u.ops.push(i);break}o[2]&&u.ops.pop(),u.trys.pop();continue}i=e.call(t,u)}catch(t){i=[6,t],r=0}finally{n=o=0}if(5&i[0])throw i[1];return{value:i[0]?i[1]:void 0,done:!0}}([i,c])}}}(this,(function(e){return this.consentsPromise||(this.consentsPromise=new Promise((function(e,o){(function(t,e){void 0===e&&(e=5e3);var o,i,u,c=f(window.__cmp)&&window.__cmp||function(){var t=v("__cmpLocator","__cmpCall");if(t)return function(e,n,r){t(e,1,r,n)}}(),s=f(window.__tcfapi)&&window.__tcfapi||v("__tcfapiLocator","__tcfapiCall"),a=p(),l=function(){if(s)return s("addEventListener",2,(function(t,e){if(e&&t&&("useractioncomplete"===t.eventStatus||"tcloaded"===t.eventStatus)){var n=void 0===u;u=t,!n&&i&&(i=null,p()),h()}})),!0}();return a||l;function p(){if(c)return c("getVendorConsents",null,(function(t,e){if(e&&t){i=t;var n=setTimeout(h,1e3);c("getConsentData",null,(function(t,e){e&&t&&(i.tcString=t.consentData,clearTimeout(n),h())}))}})),!0}function h(){!s||!c||i&&u?(o&&(clearTimeout(o),o=null),d()):o=setTimeout(d,e)}function d(){if(i||u){var e=new g;(function(t){i&&(t.tcDataV1=i,r(t.vendor,i.vendorConsents),r(t.purpose,i.purposeConsents),t.time=void 0!==i.lastUpdated?new Date(i.lastUpdated).getTime():n(i.metadata),t.tcString=i.tcString)})(e),function(t){u&&(t.tcDataV2=u,r(t.vendor,u.vendor.consents),r(t.purpose,u.purpose.consents),t.time=n(u.tcString),t.tcString=u.tcString)}(e),t(e)}}})(e,t.timeout)||o("CMP Not Found!")}))),[2,this.consentsPromise]}))},new((o=void 0)||(o=Promise))((function(n,r){function u(t){try{s(i.next(t))}catch(t){r(t)}}function c(t){try{s(i.throw(t))}catch(t){r(t)}}function s(t){var e;t.done?n(t.value):(e=t.value,e instanceof o?e:new o((function(t){t(e)}))).then(u,c)}s((i=i.apply(t,e||[])).next())}));var t,e,o,i},t}(),g=function(){function t(){this.vendor={},this.purpose={}}return t.prototype.agreementForVendor=function(t){var e=this.vendor[t],n=this.purpose;return new S(!!(e&&n[1]&&n[5]),!!(e&&n[2]&&n[3]&&n[4]),this.time||(new Date).getTime())},t.prototype.hasAgreement=function(t,e){return this.vendor[t]&&this.purpose[e]},t}(),S=function(t,e,n){this.ns1=t,this.ns2=e,this.time=n};function _(t){if(void 0===t&&(t="__atm"),l(t)){var e=window[t];if(e)return function(t){return function(e,n){return r=this,o=void 0,u=function(){return function(t,e){var n,r,o,i,u={label:0,sent:function(){if(1&o[0])throw o[1];return o[1]},trys:[],ops:[]};return i={next:c(0),throw:c(1),return:c(2)},"function"==typeof Symbol&&(i[Symbol.iterator]=function(){return this}),i;function c(i){return function(c){return function(i){if(n)throw new TypeError("Generator is already executing.");for(;u;)try{if(n=1,r&&(o=2&i[0]?r.return:i[0]?r.throw||((o=r.return)&&o.call(r),0):r.next)&&!(o=o.call(r,i[1])).done)return o;switch(r=0,o&&(i=[2&i[0],o.value]),i[0]){case 0:case 1:o=i;break;case 4:return u.label++,{value:i[1],done:!1};case 5:u.label++,r=i[1],i=[0];continue;case 7:i=u.ops.pop(),u.trys.pop();continue;default:if(!((o=(o=u.trys).length>0&&o[o.length-1])||6!==i[0]&&2!==i[0])){u=0;continue}if(3===i[0]&&(!o||i[1]>o[0]&&i[1]<o[3])){u.label=i[1];break}if(6===i[0]&&u.label<o[1]){u.label=o[1],o=i;break}if(o&&u.label<o[2]){u.label=o[2],u.ops.push(i);break}o[2]&&u.ops.pop(),u.trys.pop();continue}i=e.call(t,u)}catch(t){i=[6,t],r=0}finally{n=o=0}if(5&i[0])throw i[1];return{value:i[0]?i[1]:void 0,done:!0}}([i,c])}}}(this,(function(r){switch(e){case"getUid":case"getReadyState":case"getVersion":return[2,P(t,e,n)]}return[2,t(e,n)]}))},new((i=void 0)||(i=Promise))((function(t,e){function n(t){try{s(u.next(t))}catch(t){e(t)}}function c(t){try{s(u.throw(t))}catch(t){e(t)}}function s(e){var r;e.done?t(e.value):(r=e.value,r instanceof i?r:new i((function(t){t(r)}))).then(n,c)}s((u=u.apply(r,o||[])).next())}));var r,o,i,u}}(e)}}function P(t,e,n){var r,o=new Promise((function(t){r=t}));return n&&o.then(n),t(e,r)||o}const k=document.currentScript;var T,x=l((T=s())&&c(T,"_jtDebug")||k&&k.src&&c(k.src,"debug")||u("jtDebug"));function C(){E(arguments,"info","#00633a")}function I(){E(arguments,"error","#f02011")}function O(){E(arguments,"debug","#002db3")}function E(t,e,n,r){if(x||r){var o=[].slice.call(t);o.unshift("display: inline-block; color: #fff; background: "+n+"; padding: 1px 4px; border-radius: 3px;"),o.unshift("%cATM%c"+(o[1]&&" "+o[1]||"")),o[2]="",console[e].apply(console,o)}}const U=function(t,{request:e,done:n}={}){return function(r,i,u,c={}){try{let s,a=c.method||(u?"POST":"GET"),l=document.createElement("a");l.href=r;let h=p(i)?i:{success:function(){C("xhr success")},error:function(t){I("xhr error",null,t)}};f(i)&&(h.success=i),s=new window.XMLHttpRequest,s.onreadystatechange=function(){if(4===s.readyState){f(n)&&n(l.origin);let t=s.status;t>=200&&t<300||304===t?h.success(s.responseText,s):h.error(s.statusText,s)}},t&&(s.ontimeout=function(){I("  xhr timeout after ",s.timeout,"ms")}),"GET"===a&&u&&(r=o(r,u)),s.open(a,r,!0),t&&(s.timeout=t),c.withCredentials&&(s.withCredentials=!0),function(t,e){if(t){if(isFn(t.forEach))return t.forEach(e,this);var n=0,r=t.length;if(r>0)for(;n<r;n++)e(t[n],n);else for(n in t)hasOwnProperty.call(t,n)&&e.call(this,t[n],n)}}(c.customHeaders,((t,e)=>{s.setRequestHeader(e,t)})),c.preflight&&s.setRequestHeader("X-Requested-With","XMLHttpRequest"),s.setRequestHeader("Content-Type",c.contentType||"text/plain"),f(e)&&e(l.origin),"POST"!==a&&"PUT"!==a||!u?s.send():s.send(u)}catch(t){I("xhr construction",t)}}}();document.readyState;var j=document.currentScript;function A(t,e,n){return new Promise((function(r,o){setTimeout((function(){(e?o:r)(n)}),t)}))}var R,D=function(t,e,n,r){return new(n||(n=Promise))((function(o,i){function u(t){try{s(r.next(t))}catch(t){i(t)}}function c(t){try{s(r.throw(t))}catch(t){i(t)}}function s(t){var e;t.done?o(t.value):(e=t.value,e instanceof n?e:new n((function(t){t(e)}))).then(u,c)}s((r=r.apply(t,e||[])).next())}))},V=function(t,e){var n,r,o,i,u={label:0,sent:function(){if(1&o[0])throw o[1];return o[1]},trys:[],ops:[]};return i={next:c(0),throw:c(1),return:c(2)},"function"==typeof Symbol&&(i[Symbol.iterator]=function(){return this}),i;function c(i){return function(c){return function(i){if(n)throw new TypeError("Generator is already executing.");for(;u;)try{if(n=1,r&&(o=2&i[0]?r.return:i[0]?r.throw||((o=r.return)&&o.call(r),0):r.next)&&!(o=o.call(r,i[1])).done)return o;switch(r=0,o&&(i=[2&i[0],o.value]),i[0]){case 0:case 1:o=i;break;case 4:return u.label++,{value:i[1],done:!1};case 5:u.label++,r=i[1],i=[0];continue;case 7:i=u.ops.pop(),u.trys.pop();continue;default:if(!((o=(o=u.trys).length>0&&o[o.length-1])||6!==i[0]&&2!==i[0])){u=0;continue}if(3===i[0]&&(!o||i[1]>o[0]&&i[1]<o[3])){u.label=i[1];break}if(6===i[0]&&u.label<o[1]){u.label=o[1],o=i;break}if(o&&u.label<o[2]){u.label=o[2],u.ops.push(i);break}o[2]&&u.ops.pop(),u.trys.pop();continue}i=e.call(t,u)}catch(t){i=[6,t],r=0}finally{n=o=0}if(5&i[0])throw i[1];return{value:i[0]?i[1]:void 0,done:!0}}([i,c])}}},L=function(){function t(t){this.consentService=t}return t.prototype.customize=function(t){return D(this,void 0,void 0,(function(){var e;return V(this,(function(n){switch(n.label){case 0:return n.trys.push([0,2,,3]),e=t,[4,this.consentService.getConsents()];case 1:return e.tcString=n.sent().tcString,[3,3];case 2:return I("Can't fetch consents.",n.sent()),[3,3];case 3:return[2]}}))}))},t}(),N=function(){function t(t){this.value=t}return t.prototype.customize=function(t){return D(this,void 0,void 0,(function(){return V(this,(function(e){return t.consent=this.value,[2]}))}))},t}(),q=function(){function t(t){this.config=t}return t.prototype.customize=function(t){return D(this,void 0,void 0,(function(){var e,n,r,o,i,u;return V(this,(function(c){switch(c.label){case 0:if(e=navigator.userAgentData,!this.config.clientHintsEnabled)return[2];if(!e)return O("Client Hints Api is not present."),[2];n=e.getHighEntropyValues(["architecture","bitness","model","platform","platformVersion","uaFullVersion"]),r=e.brands.slice(-1).pop(),c.label=1;case 1:return c.trys.push([1,3,,4]),o=t,i=rt.bind,u=[void 0,r.brand,r.version,e.mobile],[4,n];case 2:return o.ch=new(i.apply(rt,u.concat([c.sent()]))),[3,4];case 3:return I("Can't get Client Hints.",c.sent()),[3,4];case 4:return[2]}}))}))},t}(),z=function(){function t(){this.isFlocEnabled=!!window.chrome&&(!!window.chrome.webstore||!!window.chrome.runtime)&&!!document.featurePolicy&&!!document.featurePolicy.features()&&document.featurePolicy.features().includes("interest-cohort")&&f(document.interestCohort)}return t.prototype.customize=function(t){return D(this,void 0,void 0,(function(){var e;return V(this,(function(n){switch(n.label){case 0:return this.isFlocEnabled?(e=t,[4,document.interestCohort()]):[3,2];case 1:e.cohortId=n.sent().id,n.label=2;case 2:return[2]}}))}))},t}(),F=function(){function t(t){void 0===t&&(t=1e4),this.timeout=t}return t.prototype.customize=function(t){return D(this,void 0,void 0,(function(){var e,n;return V(this,(function(r){switch(r.label){case 0:if(!(e=window.pbjs)||!e.que)return[3,4];r.label=1;case 1:return r.trys.push([1,3,,4]),[4,(o=new Promise((function(t){e.que.push((function(){t(e)}))})),i=this.timeout,Promise.race([o,A(i,!0,"timeout")]))];case 2:return n=r.sent(),t.pbjs=new nt(n.version,n.getUserIds()),[3,4];case 3:return O("Timout while waiting prebid",r.sent()),[3,4];case 4:return[2]}var o,i}))}))},t}(),H=function(){function t(){}return t.prototype.getStoredId=function(){return null},t.prototype.saveId=function(){},t}(),M=function(){function t(t,e){this.cookiePrefix=t,this.cookieTtl=e}return t.prototype.uidCookieName=function(){return this.cookiePrefix+"uid"},t.prototype.utCookieName=function(){return this.cookiePrefix+"ut"},t.prototype.getStoredId=function(){var t=u(this.uidCookieName()),e=+u(this.utCookieName());return t&&new G(t,e)},t.prototype.saveId=function(t,e){i(this.uidCookieName(),t,this.cookieTtl,e),i(this.utCookieName(),(new Date).getTime(),this.cookieTtl,e)},t}(),G=function(t,e){void 0===e&&(e=0),this.value=t,this.ut=e},J=(R=function(t,e){return(R=Object.setPrototypeOf||{__proto__:[]}instanceof Array&&function(t,e){t.__proto__=e}||function(t,e){for(var n in e)Object.prototype.hasOwnProperty.call(e,n)&&(t[n]=e[n])})(t,e)},function(t,e){if("function"!=typeof e&&null!==e)throw new TypeError("Class extends value "+String(e)+" is not a constructor or null");function n(){this.constructor=t}R(t,e),t.prototype=null===e?Object.create(e):(n.prototype=e.prototype,new n)}),X=function(t,e,n,r){return new(n||(n=Promise))((function(o,i){function u(t){try{s(r.next(t))}catch(t){i(t)}}function c(t){try{s(r.throw(t))}catch(t){i(t)}}function s(t){var e;t.done?o(t.value):(e=t.value,e instanceof n?e:new n((function(t){t(e)}))).then(u,c)}s((r=r.apply(t,e||[])).next())}))},W=function(t,e){var n,r,o,i,u={label:0,sent:function(){if(1&o[0])throw o[1];return o[1]},trys:[],ops:[]};return i={next:c(0),throw:c(1),return:c(2)},"function"==typeof Symbol&&(i[Symbol.iterator]=function(){return this}),i;function c(i){return function(c){return function(i){if(n)throw new TypeError("Generator is already executing.");for(;u;)try{if(n=1,r&&(o=2&i[0]?r.return:i[0]?r.throw||((o=r.return)&&o.call(r),0):r.next)&&!(o=o.call(r,i[1])).done)return o;switch(r=0,o&&(i=[2&i[0],o.value]),i[0]){case 0:case 1:o=i;break;case 4:return u.label++,{value:i[1],done:!1};case 5:u.label++,r=i[1],i=[0];continue;case 7:i=u.ops.pop(),u.trys.pop();continue;default:if(!((o=(o=u.trys).length>0&&o[o.length-1])||6!==i[0]&&2!==i[0])){u=0;continue}if(3===i[0]&&(!o||i[1]>o[0]&&i[1]<o[3])){u.label=i[1];break}if(6===i[0]&&u.label<o[1]){u.label=o[1],o=i;break}if(o&&u.label<o[2]){u.label=o[2],u.ops.push(i);break}o[2]&&u.ops.pop(),u.trys.pop();continue}i=e.call(t,u)}catch(t){i=[6,t],r=0}finally{n=o=0}if(5&i[0])throw i[1];return{value:i[0]?i[1]:void 0,done:!0}}([i,c])}}},B=function(){function t(t){this.uid=t.getUid()}return t.prototype.getUid=function(){return this.uid},t}(),K=function(){function t(t){this.baseIdService=t}return t.prototype.getUid=function(){return X(this,void 0,void 0,(function(){var t;return W(this,(function(e){switch(e.label){case 0:return[4,this.baseIdService.getUid()];case 1:return l(t=e.sent())&&j.dispatchEvent(new CustomEvent("justIdReady",{detail:{justId:t}})),[2,t]}}))}))},t}(),Q=function(){function t(){}return t.prototype.getUid=function(){return Promise.resolve(null)},t}(),Y=function(){function t(){}return t.prototype.getUid=function(){return Promise.resolve(t.getDebugJustId())},t.shouldUse=function(){return l(t.getDebugJustId())},t.getDebugJustId=function(){var t=s();return t&&new URL(t).searchParams.get("__jtUid")},t}(),Z=function(){function t(t){this.atm=t}return t.prototype.getUid=function(){return this.atm("getUid")},t}(),$=function(){function t(t,e){this.config=t,this.customizers=[],this.customizers.push(new q(t)),this.customizers.push(new z),this.customizers.push(new F(t.pbjsTimeout)),this.customizers.push(t.tcfRequired?new L(e):new N(!0)),this.idStore=t.cookiePrefix?new M(t.cookiePrefix,t.cookieTtl):new H}return t.prototype.getUid=function(){return X(this,void 0,void 0,(function(){var t,e,n;return W(this,(function(r){switch(r.label){case 0:return(t=this.idStore.getStoredId())&&t.ut+1e3*this.config.cookieRefresh>(new Date).getTime()?[2,t.value]:[3,1];case 1:return[4,this.callApi(t)];case 2:if(e=r.sent())return n=e.uid,this.idStore.saveId(n,e.tld),[2,n];r.label=3;case 3:return[2]}}))}))},t.prototype.callApi=function(t){var e=this;return new Promise((function(n,r){var o={success:function(t,e){if(200==e.status){var r=JSON.parse(t);C("IdResponse",r,e),n(r)}else n(null)},error:function(t,e){I("id err",t,e),r(t)}};e.prepareRequest(t).then((function(t){U(e.config.url,o,JSON.stringify(t),{withCredentials:!0,method:"POST"})}))}))},t.prototype.prepareRequest=function(t){return X(this,void 0,void 0,(function(){var e;return W(this,(function(n){switch(n.label){case 0:return e=new et(null==t?void 0:t.value,s(),function(){try{return window.top.document.referrer}catch(t){}}(),navigator.userAgent,function(){try{return window.top.location.href,!0}catch(t){return!1}}()),[4,Promise.all(this.customizers.map((function(t){return t.customize(e)})))];case 1:return n.sent(),[2,e]}}))}))},t}();function tt(t){if(t&&t.length>0)return t}var et=function(t,e,n,r,o){this.prevStoredId=t,this.url=e,this.referrer=n,this.userAgent=r,this.topLevelAccess=o,this.clientLib="atm"},nt=function(t,e){this.version=t,this.uids=e},rt=function(t){function e(e,n,r,o){var i=t.call(this,tt(o.architecture),tt(o.model),tt(o.platform),tt(o.platformVersion),tt(o.uaFullVersion))||this;return i.brand=e,i.brandVer=n,i.mobile=r,i}return J(e,t),e}((function(t,e,n,r,o){this.architecture=t,this.model=e,this.platform=n,this.platformVersion=r,this.uaFullVersion=o})),ot=function(){var t=function(e,n){return(t=Object.setPrototypeOf||{__proto__:[]}instanceof Array&&function(t,e){t.__proto__=e}||function(t,e){for(var n in e)Object.prototype.hasOwnProperty.call(e,n)&&(t[n]=e[n])})(e,n)};return function(e,n){if("function"!=typeof n&&null!==n)throw new TypeError("Class extends value "+String(n)+" is not a constructor or null");function r(){this.constructor=e}t(e,n),e.prototype=null===n?Object.create(n):(r.prototype=n.prototype,new r)}}(),it=function(t){function e(e,n,r){var o=t.call(this,n,r)||this;return o.atmVarName=e,o}return ot(e,t),e.prototype.getIdServiceInternal=function(){var e=_(this.atmVarName);return e?new Z(e):t.prototype.getIdServiceInternal.call(this)},e}(function(t){function e(e,n){var r=t.call(this)||this;return r.idServerConfig=e,r.consentService=n,r}return ot(e,t),e.prototype.getIdServiceInternal=function(){return this.idServerConfig&&this.idServerConfig.url?new $(this.idServerConfig,this.consentService):new Q},e}(function(){function t(){}return t.prototype.getIdService=function(){return new B(new K(Y.shouldUse()?new Y:this.getIdServiceInternal()))},t}()));function ut(t){new it(t.atmVarName,t.idServer,new b).getIdService().getUid()}GET_ID=e})();
GET_ID.init({"idServer":{"url":"https://id.qpa.audience-solutions.com/getId","cookiePrefix":"__jt","cookieTtl":63072000,"cookieRefresh":28800,"clientHintsEnabled":true,"pbjsTimeout":10000,"tcfRequired":true}});
})();
} catch(e) {}`

Reply:
I took your comments into account